src/ConfiguredMediator.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Copyright by the ComposerDistributor-Team
+ *
+ * Licenses under the MIT-license. For details see the included file LICENSE.md
+ */
+
+declare(strict_types=1);
+
+namespace PharIo\ComposerDistributor;
+
+use Composer\Installer\PackageEvent;
+use PharIo\ComposerDistributor\Config\Config;
+use PharIo\ComposerDistributor\Config\Phar;
+
+abstract class ConfiguredMediator extends PluginBase
+{
+	protected abstract function getMediatorConfig(): string;
+
+	public function installOrUpdateFunction(PackageEvent $event): void
+	{
+		$config    = $this->createConfig();
+		$installer = $this->createInstaller(
+			$config->package(),
+			$config->keyDirectory(),
+			$event
+		);
+
+		$files = [];
+		foreach ($config->phars() as $phar) {
+			$files[] = new File(
+				$phar->name(),
+				Url::fromString($phar->file()),
+				Url::fromString($phar->signature())
```

Comment:
Are these always URLs? If so, we could move these calls into the Config and only return URLs rather than plain strings. That would avoid the need to handle exceptions from `Url::fromString` in the case the string value is b0rked...

Reply:
I can't think of any case where something other than an URL would make sense, I'll move it to the configuration.

Reply:
Done